src/backend/distributed/shared_library_init.c

```diff
@@ -1792,6 +1795,28 @@ NoticeIfSubqueryPushdownEnabled(bool *newval, void **extra, GucSource source)
 }
 
 
+/*
+ * WarnIfReplicationModelIsSet prints a warning when a user sets
+ * citus.replication_model.
+ */
+static bool
+WarnIfReplicationModelIsSet(int *newval, void **extra, GucSource source)
+{
+	ereport(NOTICE, (errcode(ERRCODE_WARNING_DEPRECATED_FEATURE),
```

Comment:
I think this notice is too aggressive, e.g. it shows up immediately when I do psql just because I have it in my postgresql.conf.
```
$ psql
NOTICE:  Setting citus.replication_model has no effect. Please use citus.shard_replication_factor instead.
DETAIL:  Citus determines the replication model based on the replication factor and the replication models of the colocated shards. If a colocated table is present, the replication model is inherited. Otherwise 'streaming' replication is preferred if supported by the replication factor.
```

Reply:
Maybe check for `source == PGC_S_SESSION`

Reply:
Now we check if the source of the GUC setting is from a query run by the user so that `.psqlrc` or `postgresql.conf` settings will not generate warnings